core/src/backends/eclair/rest/types.rs

```diff
@@ -84,6 +87,101 @@ pub enum ChannelState {
     Pending,
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Chain {
+    pub chain: String,
+    pub network: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Feature {
+    pub name: String,
+    pub is_required: bool,
+    pub is_known: bool,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InvoiceResponse {
+    pub payment_request: PayementRequestResponse,
+    pub payment_preimage: Option<String>,
+    pub status: InvoiceStatusResponse,
+    pub created_at: InvoiceDateResponse,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PayementRequestResponse {
+    pub prefix: String,
+    pub timestamp: i64,
+    pub node_id: String,
+    pub serialized: String,
+    pub description: String,
+    pub payment_hash: String,
+    pub payment_metadata: String,
+    pub expiry: u64,
+    pub min_final_cltv_expiry: u32,
+    pub amount: u64,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum InvoiceStatusTypeResponse {
+    Received,
+    Pending,
+    Expired,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct InvoiceDateResponse {
+    pub unix: u64,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct InvoiceStatusResponse {
+    // "type" is a reserved keyword. Fix it with an _
+    pub _type: InvoiceStatusTypeResponse,
```

Comment:
An other cool way of doing this (open to discussion):
```suggestion
    #[serde(rename = "type")]
    pub type_field: InvoiceStatusTypeResponse,
```
Mainly because the underscore is generally reserved for private / unused fields / params.

Reply:
Ah great!